var/spack/repos/builtin/packages/embree/package.py

```diff
@@ -48,17 +49,19 @@ def cmake_args(self):
         # code selection and defines controlling namespace names are based on
         # defines controlled by compiler flags, so disable ISAs below compiler
         # flags chosen by spack
-        if spec.target >= 'nehalem':
+        if spec.target >= 'x86_64_v2':
             args.append('-DEMBREE_ISA_SSE2=OFF')
         else:
             args.append('-DEMBREE_ISA_SSE2=ON')
 
-        if spec.target >= 'sandybridge':
+        if spec.target >= 'sandybridge' \
+                or spec.target >= 'bulldozer' \
+                or spec.target >= 'x86_64_v3':
             args.append('-DEMBREE_ISA_SSE42=OFF')
         else:
             args.append('-DEMBREE_ISA_SSE42=ON')
 
-        if spec.target >= 'haswell':
+        if spec.target >= 'x86_64_v3':
```

Comment:
I think this can be written as:
```python
self.define('EMBREE_ISA_AVX', 'avx' in spec.target)
```
similarly for the other defines above.

Reply:
Oh yeah, this simplifies things massively! Thanks!